services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/helpers/Wrappers.groovy

```diff
@@ -34,6 +36,23 @@ class Wrappers {
         throw new MultipleFailureException(ex)
     }
 
+    /**
+     * Executes given closure in a loop for certain amount of time.
+     * If time is less or equal to 0, the closure won't be executed even once.
+     * Loop may take a bit longer than 'timeout' by maximum of closure execution time, since check is performed
+     * before the closure call.
+     *
+     * @param timeout For how long to execute the given closure in a loop
+     * @param unit In which unit the 'timeout' value is specified
+     * @param closure code to repeat in a loop
+     */
+    static void timedLoop(long timeout, TimeUnit unit = TimeUnit.SECONDS, Closure closure) {
```

Comment:
1. Can we rename it to just 'loop' to have consistency with 'wait' and 'retry'? I don't insist, it is just a question.
2. I am not sure about timeout as long.  For example, I will want to execute my closure for 3.5 mins. I think we need to use double here.

Reply:
TimeUnit only understands long. You'll have to pass 3500 milliseconds instead. Otherwise, if I change to double I'll have to remove the ability to select 'unit' and always require seconds. Do you prefer to change this way?

Reply:
A self-made min->sec->millis converter for double values would be an overkill I think

Reply:
I don't think that we need to leave the possibility to choose a unit for this method. As far as I understand we can pass milliseconds, seconds, minutes, etc. But milliseconds unlikely will be used. Minutes can be used but only integer values. So my point is to use just seconds as in wait.

Reply:
Changed similar to 'wait' to always accept seconds